src/legacy/core_plugins/kibana/public/discover/helpers/build_services.ts

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import {
+  Capabilities,
+  ChromeStart,
+  CoreStart,
+  DocLinksStart,
+  ToastsStart,
+  UiSettingsClientContract,
+} from 'kibana/public';
+import * as docViewsRegistry from 'ui/registry/doc_views';
+import chromeLegacy from 'ui/chrome';
+import { IPrivate } from 'ui/private';
+import { FilterManager, TimefilterContract } from 'src/plugins/data/public';
+// @ts-ignore
+import { StateProvider } from 'ui/state_management/state';
+// @ts-ignore
+import { createSavedSearchesService } from '../saved_searches/saved_searches';
+// @ts-ignore
+import { createSavedSearchFactory } from '../saved_searches/_saved_search';
+import { DiscoverStartPlugins } from '../plugin';
+import { start as legacyData } from '../../../../data/public/legacy';
+import { IndexPatterns } from '../../../../data/public/index_patterns/index_patterns';
```

Comment:
Please refrain from importing deep withing NP plugins.
Same thing is exported from `../../../../data/public`

Reply:
I think if you add this inside a `np_ready` folder, this stuff will be linted for you.  That right @lizozom?

Reply:
Merci! done, dear @stacey-gammon did you find time to have a look at the embeddable part of this PR? thx!

Reply:
I took a very quick look at the embeddable changes.  I think it looks fine, at least nothing bad stood out, but I haven't had a chance to play with the code, or really understand the reasons for the changes (is the goal to get rid of `getServices` entirely?.. I see it was replaced by passed down functionality which seems to be the point, but there are still other `getServices` calls, so maybe it's not all shimmed yet?

Not sure I'll have time to pull code & test today, but if you get enough reviews, don't let me hold the PR up!

Reply:
Thx Stacey, currently getServices is necessary, but it should definitely go away in future iterations.